### PR TITLE
Create visible_label locator to maintain backward compatibility

### PR DESCRIPTION
**Pull request by titusfortner**
This is an alternative to #717 

I don't love this locator, but I think I like this locator more than I like the alternatives (breaking backward compatibility and nested hashes).

### CODE REVIEW

**Review by p0deje — APPROVED** [review]
I also don't quite like that solution, but I guess it's the only way.

**Comment by sjieg** [comment]
Yes! This makes much more sense:
```
expect(browser.text_field(label: 'With hidden text')).to exist
expect(browser.text_field(label: 'With text')).not_to exist
expect(browser.text_field(visible_label: /With text/)).to exist
expect(browser.text_field(visible_label: /With hidden text/)).not_to exist
```
I've closed my PR.

File: lib/watir/locators/element/locator.rb [hunk 1]
```diff
@@ -165,10 +171,10 @@ def create_filter_selector
           @filter_selector
         end
 
-        def process_label
-          return unless @normalized_selector[:label].kind_of?(Regexp) && selector_builder.should_use_label_element?
+        def process_label(label_key)
+          return unless @normalized_selector[label_key].kind_of?(Regexp) && selector_builder.should_use_label_element?
```
Comment (jkotests) [inline lib/watir/locators/element/locator.rb]: This check will need to be different based on the label key:
* :label is only processed if a Regexp (no change)
* :visible_label is always processed - ie regardless of whether String or Regexp

Right now, `browser.text_field(visible_label: 'text')` will go into the XPath builder and be treated like a custom attribute.

**Review by jkotests — CHANGES_REQUESTED** [review]
I'm still not a fan. I agree a nicer looking API than nested Hashes, but a nuisance in the code base.

File: lib/watir/locators/element/locator.rb (hunk 1 above)
Comment (titusfortner) [reply lib/watir/locators/element/locator.rb]: ugh, you're right. sigh.

**Comment by titusfortner** [comment]
@jkotests  the fix was simpler than I feared. I'm thinking we don't even advertise this locator, and only mention it in the deprecation notice for the handful of people it might apply to.

**Comment by jkotests** [comment]
@titusfortner , that sounds reasonable to me.

**Review by jkotests — APPROVED** [review]

**Pull request merged**